tutorials/install-pgbouncer/index.mdx

```diff
@@ -0,0 +1,252 @@
+
+## Recommended configurations
+
+- The pooling method can be configured both for the database and globally. (Default: session)
+   ```
+   pool_mode = <session, transaction or statement>
+   ```
+
+- This configuration shows the maximum number of client connections allowed to PgBouncer. You may want to set it large because it can block your connection request. The best practice is to limit this with `pool_size`. Be aware that if you are planning to reach this limit, you may want to increase the number of file descriptors. (Default: 100)
+   ```
+   max_client_conn = max_client_conn + (max pool_size * total databases * total users)
+   ```
+
+- You may want to increase this number to 5 or 10. When the specified `pool_size` is not enough, it uses the reserved pool and logs it. It can be used to determine `pool_size`. (Default: 0)
+   ```
+   reserve_pool_size = 10
+   ```
+
+- Connecting to PostgreSQL through PgBouncer will mask the PostgreSQL IP. This setting adds the client host address and port to the application name. It is very helpful when troubleshooting. (Default: 0)
+   ```
+   application_name_add_host = 1
+   ```
+
+- By default, PgBouncer reuses server connections in LIFO (last-in, first-out). If you are using a TCP load balancer with round-robin behind the PostgreSQL IP address, you may want to enable `server_round_robin` for achieve higher performance. (Default: 1)
+   ```
+   server_round_robin = 1
+   ```
+
+## Recommended authentication method
+By default, PgBouncer comes with `trust` authentication method. The `trust` method can used for testing environment but is not recommended for production. It will accept all connection requests without authentication. For production, `hba` authentication is recommended. With this, you can specify IP addresses with different connection methods:
+
+### HBA Authentication (Recommended for production):
+
+1. Change the `auth_type` and add the `auth_hba_file` path:
+   ```
+   [pgbouncer]
+   auth_type = hba
+   auth_hba_file = /etc/pgbouncer/pb_hba.conf
+   ```
+
+2. Create the `/etc/pgbouncer/pb_hba.conf` file with the following content:
+   ```   
+   # TYPE DATABASE USER ADDRESS METHOD
+
+   # IPv4 local connections:
+   host all all 127.0.0.1/32 scram-sha-256
+
+   # IPv6 local connections:
+   host all all ::1/128 scram-sha-256
+
+   # Application
+   host all all APPLICATION_IP/NETMASK scram-sha-256
+   ```
+
+3. Create the `/etc/pgbouncer/userlist.txt` file that contains the user allowed to login from PgBouncer:
+   ```
+   "username1" "password"
+   "username2" "md5abcdef012342345"
+   "username2" "SCRAM-SHA-256$<iterations>:<salt>$<storedkey>:<serverkey>"
+    ```
+
+    <Message type="tip">
+
+    You can find the hashed passwords in `pg_shadow` table in PostgreSQL.
+    ```
+    SELECT usename,passwd FROM pg_shadow;
+    ```
+
+    </Message>
+
+4. Add the IP address of the PgBouncer server to the PostgreSQL `pg_hba.conf` file:
+   ```
+   host all all PGBOUNCER_IP/NETMASK trust
```

Comment:
with hba config this can be switched to `md5` or `scram-sha-256` instead of `trust`

Reply:
You are right. Thanks. :)

Fixed in b60d4b8a8395528d169a4e16e124ed611e327cbb